dev/MANUALLY_GENERATING_IMAGE_CACHE_AND_CONSTRAINTS.md

```diff
@@ -532,3 +359,196 @@ breeze release-management update-constraints --constraints-repo /home/user/airfl
     --commit-message "Update pymssql constraint to 2.2.8 and Authlib to 1.3.0" \
     --airflow-constraints-mode constraints
 ```
+
+
+# Figuring out backtracking dependencies
+
+## Why we need to figure out backtracking dependencies
+
+Sometimes, very rarely the CI image in `canary` builds take a very long time to build. This is usually
+caused by `pip` trying to figure out the latest set of dependencies (`eager upgrade`) .
+The resolution of dependencies is a very complex problem and sometimes it takes a long time to figure out
+the best set of dependencies. This is especially true when we have a lot of dependencies and they all have
+to be found compatible with each other. In case new dependencies are released, sometimes `pip` enters
+a long loop trying to figure out if the newly released dependency can be used, but due to some other
+dependencies of ours it is impossible, but it will take `pip` a very long time to figure it out.
```

Comment:
```suggestion
to be found compatible with each other. In case new dependencies are released, `pip` might enter
a long loop trying to figure out if these dependencies can be used. Unfortunately, this long loop could end up in an error due to conflicts.
```